core/input/input.h

```diff
@@ -53,6 +81,28 @@ class Input : public Object {
 		MOUSE_MODE_CONFINED_HIDDEN,
 	};
 
+	enum BufferingMode {
+		BUFFERING_MODE_NONE,
+		BUFFERING_MODE_FRAME,
+		BUFFERING_MODE_AGILE,
+	};
+
+protected:
+	struct Data {
+		BufferingMode buffering_mode = BUFFERING_MODE_FRAME;
+		bool use_accumulated_input = true;
+		bool use_buffering = false;
+		bool use_agile = false;
+		bool use_legacy_flushing = false;
+		bool has_input_thread = false;
```

Comment:
Maybe bitfields or flags?

Reply:
I'm usually an advocate of using bitwise flags, but I'm not convinced there is a super pressing need here:

* Memory - this is only stored once for the whole app
* Speed - as far as I remember it is only performed rarely

It probably doesn't have a massive effect on readability either way.
That said I'm happy to change if in general this is preferred. :slightly_smiling_face: